internal/presenter/packages/spdx_json_presenter.go

```diff
@@ -41,12 +43,16 @@ func (pres *SPDXJsonPresenter) Present(output io.Writer) error {
 
 // newSPDXJsonDocument creates and populates a new JSON document struct that follows the SPDX 2.2 spec from the given cataloging results.
 func newSPDXJsonDocument(catalog *pkg.Catalog, srcMetadata source.Metadata) spdx22.Document {
-	var name string
+	uID := uuid.Must(uuid.NewRandom())
+
+	var name, namespace string
 	switch srcMetadata.Scheme {
 	case source.ImageScheme:
 		name = srcMetadata.ImageMetadata.UserInput
+		namespace = strings.Trim(fmt.Sprintf("%s-%s", name, uID.String()), "-")
 	case source.DirectoryScheme:
 		name = srcMetadata.Path
+		namespace = uID.String()
```

Comment:
Why do we construct this value using two different approaches here?

Reply:
Good question - I opted to keep the `ImageMetadata.UserInput` as the namespace prefix for images since that's what exists on main. 

For the directory, I could not think of a great metadata identifier besides the digested contents path. The path presents a few challenges when appending it to our `https://anchore.com/syft/image/%s` format.

A user could specify an absolute path `/foo/bar` which would result in `...image//foo/bar-<uuid>`.
A user could also specify a relative path `./foo/bar` which would result in `...image/./foo/bar-<uuid>`

Rather than parse/clean all possible `fs` inputs I decided to opt for simplicity and generate a uuid since the `documentNamespace` field on main for a directory input is currently not set.

Reply:
That's a good callout. 

This behavior also exists currently on `main` so I'm happy to use this branch to squash this bug. I was able to reproduce it on my machine with alpine and saw this generated with the latest version of syft:

`"documentNamespace": "https://anchore.com/syft/image//Users/hal/development/images/alpine",`

We could run the resulting `namespace` string through a cleaner function that strips all bad prefixes.

We could also adopt a common method of just appending a `uuid` since the sbom have other fields that provide context as to what was scanned.

Reply:
These are good ideas. I don't have a strong suggestion here. Here's what I do think...

1. We should probably use the same logic for building this string, regardless of an image or directory path. This lets us factor out this string generation as a function, if we want. Also, we should probably remove "image" from `"https://anchore.com/syft/image"`, since this value is getting used in both paths (this is true both _already_ and potentially _going forward_, too).

2. We could go the route of cleaning the user input value, but I could also see that potentially getting hairy. It'd be great if there was a convention we could lean on that exists already — one that has accounted for producing safe "job identifying" strings. I'm not sure if this exists today. So perhaps for now, UUIDs alone are enough (instead of incorporating user input)?

Reply:
Yea - I think UUIDs alone are good enough until we start seeing more of a standard being adopted across the sbom community. I'll also update the path so we have `syft/dir` vs `syft/image` depending on the input contents